accepted/0000-reduce-lifecycle-script-environment.md

```diff
@@ -0,0 +1,129 @@
+# Reduce the environment variables provided to lifecycle scripts
+
+In versions of npm up to v6, the following items are all placed into the
+environment of scripts run for various lifecycle events (install, prepare,
+etc., as well as explicit scripts such as test and start).
+
+- `PATH` Configured to include `.../node_modules/.bin` for current and all
+  parent `node_modules` directories.
+- `npm_package_*` for all `package.json` values in the current package for
+  which the lifecycle event is running.
+- `npm_config_*` for all npm configuration values that do not start with a
+  `_` character.
+- `npm_lifecycle_event` the current lifecycle event.
+- `npm_lifecycle_script` the command being run.
+- `npm_node_execpath` the path to the Node.js executable npm is using.
+- `npm_execpath` the path to the npm executable being run.
+
+The suggestion presented here is to remove (or vastly reduce) the
+`npm_config_*` and `npm_package_*` environment variables from the context
+of lifecycle scripts, and potentiaally also add new fields that may be more
```

Comment:
In particular, i use some of these fields in https://npmjs.com/safe-publish-latest in "prepublish"/"prepare"/"prepublishOnly", and also in "version"/"preversion"/"postversion" in a few packages. Hopefully the vars that are relevant to the particular script can be preserved?

```suggestion
of lifecycle scripts, and potentially also add new fields that may be more
```

Reply:
@ljharb latest version of `in-publish` looks at the `npm_config` env. https://github.com/iarna/in-publish/pull/14

Reply:
Thanks! With that, and assuming [`npm_config_tag`](https://github.com/ljharb/safe-publish-latest/blob/632c8a9c613a5c1a631e64cef86f5f3ab22b94cc/getTag.js#L4) is still available in `prepublish` scripts, everything should continue working properly ❤️

Reply:
Well, it'll be included if it's changed.  Otherwise you'll have to just know that the default is `latest`.

Reply:
Oh, nevermind, I see that you're doing that in the line you linked.  All good then, yes, that should work.